### PR TITLE
SOLR-17019: ZkCli should create subpaths when necessary

### PR DESCRIPTION
**Pull request by HoustonPutman**
https://issues.apache.org/jira/browse/SOLR-17019

### CODE REVIEW

**Review by risdenk — APPROVED** [review]
Do we need to worry about ACLs on these created znodes?

**Comment by HoustonPutman** [comment]
> Do we need to worry about ACLs on these created znodes?

Good question! So zkCli.sh take the `SOLR_ZK_CREDS_AND_ACLS` envVar, just like the `bin/solr` tool commands do. So both should use the same ACLs when starting the Solr ZK Client.

File: solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ClusterProperties.java [hunk 1]
```diff
@@ -220,7 +220,7 @@ public void setClusterProperty(String propertyName, Object propertyValue) throws
           @SuppressWarnings({"rawtypes"})
           Map properties = new LinkedHashMap();
           properties.put(propertyName, propertyValue);
-          client.create(
+          client.makePath(
```
Comment (stillalex) [inline solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ClusterProperties.java:223]: minor: this doesn't seem to need the change, given that it's a constant.

**Review by stillalex — APPROVED** [review]

File: solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ClusterProperties.java (hunk 1 above)
Comment (HoustonPutman) [reply solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ClusterProperties.java:223]: This is what necessitated the change. the `chroot` that the user is using might not exist yet.

**Comment by epugh** [comment]
I don't know if this matters at all, but at some point, I'd like to see what `zkCli.sh` does be subsumed into `bin/solr zk` commands.   Obviously with the appropriate deprecation process...

**Comment by HoustonPutman** [comment]
> I don't know if this matters at all, but at some point, I'd like to see what `zkCli.sh` does be subsumed into `bin/solr zk` commands. Obviously with the appropriate deprecation process...

Yeah, I'm not a fan of the separate utilities. One path for all this stuff would be much better. Anyways, this would need to be fixed either way.

File: solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ClusterProperties.java (hunk 1 above)
Comment (stillalex) [reply solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ClusterProperties.java:223]: yep. wrote the comment and like 10 mins later...oh yeah... thanks for confirming :)

**Pull request merged**